script.js

```diff
@@ -1,12 +1,225 @@
+  return countEl;
+}
+// Writing the name and number in correct syntax
+function displayNameNumber(episode) {
+  let formattedName;
+  let episodeNum = episode.number;
+  let seasonNum = episode.season;
+
+  if (episodeNum < 10) {
+    episodeNum = `0${episode.number}`;
+  }
+  if (seasonNum < 10) {
+    seasonNum = `0${episode.season}`;
+  }
+  if (episodeNum || seasonNum) {
+    formattedName = `S${seasonNum}E${episodeNum} - ${episode.name}`;
+  } else {
+    formattedName = episode.name;
+  }
+  return formattedName;
+}
+
+// Search Bar event Listener
+searchBar.addEventListener("input", (e) => {
+  e.preventDefault();
+  //Getting value of searchInput
+  let searchString = e.target.value.toLowerCase();
+  let searchResult = search(searchString, allEpisodes); // our search result will be input text checked in all episodes
+  populateCards(searchResult);
+  showCount(searchResult);
+});
+
+// function to check if searchInput matches episodes
+// It takes two parameters inputText and Episodes we need to perform search on(in this case all episodes)
+function search(searchText, allEpisodes) {
+  let filteredEpisodes = allEpisodes.filter((episode) => {
+    if (episode.summary != undefined) {
+      // Because some shows/episodes does not have a defined summary. So we used a condition for that.
+      return (
+        episode.name.toLowerCase().includes(searchText) ||
+        episode.summary.toLowerCase().includes(searchText)
+      );
+    } else {
+      return episode.name.toLowerCase().includes(searchText);
+    }
+  });
+  return filteredEpisodes;
+}
+
+// Create a single Option for a show/
+function createOneShowOption(show) {
+  let option = document.createElement("option");
+  option.setAttribute("value", show.id);
+  option.innerHTML = show.name;
+  return option;
 }
 
-function makePageForEpisodes(episodeList) {
-  const rootElem = document.getElementById("root");
-  rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+// Create options for all shows/
+// First we appended an option to show dropdown and we've set its a special Id and Name.
+// Then we created option for each show as per above function "createOneShowOption(show)"
+function createAllShowsOptions(allShows) {
+  showDropDown.appendChild(
+    createOneShowOption({ name: "All-Shows", id: "All-Shows" }) // We are setting a default Id and name for this one option
+  );
+  allShows.forEach((show) => {
+    let showOption = createOneShowOption(show);
+    showDropDown.appendChild(showOption);
+  });
 }
 
-window.onload = setup;
+// Show DropDown Event Listener
+showDropDown.addEventListener("change", (e) => {
+  let showId = e.target.value;
+  if (showId === "All-Shows") {
+    // Should display All shows
+    populateCards(allShows);
+    removeShowCount();
+    createAllEpisodesOptions([
+      {
+        name: "Show selection required",
+        id: "Show selection required",
+      },
+    ]);
+
+    searchBar.disabled = true; // Keeps searchBar disabled
```

Comment:
I didn't know this was a thing! I disabled mine by just hiding it by setting display to none. This is super useful and I might steal, thank you thank you.

Reply:
It's an honour :)